diff -up protobuf-c-1.5.0/configure.ac.omv~ protobuf-c-1.5.0/configure.ac
--- protobuf-c-1.5.0/configure.ac.omv~	2024-03-22 19:20:35.476884393 +0100
+++ protobuf-c-1.5.0/configure.ac	2024-03-22 19:20:41.552959977 +0100
@@ -69,7 +69,7 @@ AC_ARG_ENABLE([protoc],
   AS_HELP_STRING([--disable-protoc], [Disable building protoc_c (also disables tests)]))
 if test "x$enable_protoc" != "xno"; then
   AC_LANG_PUSH([C++])
-  AX_CXX_COMPILE_STDCXX(17, noext, mandatory)
+  AX_CXX_COMPILE_STDCXX(20, noext, mandatory)
   PKG_CHECK_MODULES([protobuf], [protobuf >= 3.0.0])
 
   save_CPPFLAGS="$CPPFLAGS"
