diff -up protobuf-c-1.4.1/protoc-c/c_bytes_field.cc.omv~ protobuf-c-1.4.1/protoc-c/c_bytes_field.cc
diff -up protobuf-c-1.4.1/protoc-c/c_enum.h.omv~ protobuf-c-1.4.1/protoc-c/c_enum.h
--- protobuf-c-1.4.1/protoc-c/c_enum.h.omv~	2023-05-09 18:02:46.416222248 +0200
+++ protobuf-c-1.4.1/protoc-c/c_enum.h	2023-05-09 18:03:11.205475156 +0200
@@ -64,7 +64,11 @@
 #define GOOGLE_PROTOBUF_COMPILER_C_ENUM_H__
 
 #include <string>
+#include <sstream>
+// Yes, this is evil. You're free to fix it. Please share the patch.
+#define private public
 #include <google/protobuf/descriptor.h>
+#undef private
 
 #undef GOOGLE_DISALLOW_EVIL_CONSTRUCTORS
 #define GOOGLE_DISALLOW_EVIL_CONSTRUCTORS(TypeName) \
diff -up protobuf-c-1.4.1/protoc-c/c_field.h.omv~ protobuf-c-1.4.1/protoc-c/c_field.h
--- protobuf-c-1.4.1/protoc-c/c_field.h.omv~	2023-05-09 18:00:00.463536468 +0200
+++ protobuf-c-1.4.1/protoc-c/c_field.h	2023-05-09 18:02:00.904755211 +0200
@@ -64,8 +64,12 @@
 #define GOOGLE_PROTOBUF_COMPILER_C_FIELD_H__
 
 #include <memory>
+#include <sstream>
 #include <google/protobuf/stubs/common.h>
+// Yes, this is evil. You're free to fix it. Please share the patch.
+#define private public
 #include <google/protobuf/descriptor.h>
+#undef private
 
 #undef GOOGLE_DISALLOW_EVIL_CONSTRUCTORS
 #define GOOGLE_DISALLOW_EVIL_CONSTRUCTORS(TypeName) \
diff -up protobuf-c-1.4.1/protoc-c/c_helpers.h.omv~ protobuf-c-1.4.1/protoc-c/c_helpers.h
--- protobuf-c-1.4.1/protoc-c/c_helpers.h.omv~	2023-05-09 17:57:52.556348322 +0200
+++ protobuf-c-1.4.1/protoc-c/c_helpers.h	2023-05-09 18:04:16.099132702 +0200
@@ -66,7 +66,10 @@
 #include <string>
 #include <vector>
 #include <sstream>
+// Yes, this is evil. You're free to fix it. Please share the patch.
+#define private public
 #include <google/protobuf/descriptor.h>
+#undef private
 #include <protobuf-c/protobuf-c.pb.h>
 #include <google/protobuf/io/printer.h>
 
diff -up protobuf-c-1.4.1/protoc-c/c_service.h.omv~ protobuf-c-1.4.1/protoc-c/c_service.h
--- protobuf-c-1.4.1/protoc-c/c_service.h.omv~	2023-05-09 21:05:30.115426875 +0200
+++ protobuf-c-1.4.1/protoc-c/c_service.h	2023-05-09 21:05:42.520534486 +0200
@@ -65,7 +65,11 @@
 
 #include <map>
 #include <string>
+#include <sstream>
+// Yes, this is evil. You're free to fix it. Please share the patch.
+#define private public
 #include <google/protobuf/descriptor.h>
+#undef private
 
 #undef GOOGLE_DISALLOW_EVIL_CONSTRUCTORS
 #define GOOGLE_DISALLOW_EVIL_CONSTRUCTORS(TypeName) \
diff -up protobuf-c-1.4.1/protoc-c/c_generator.cc.omv~ protobuf-c-1.4.1/protoc-c/c_generator.cc
--- protobuf-c-1.4.1/protoc-c/c_generator.cc.omv~	2023-08-12 15:43:32.050134095 +0200
+++ protobuf-c-1.4.1/protoc-c/c_generator.cc	2023-08-12 15:44:35.157785910 +0200
@@ -60,6 +60,12 @@
 
 // Modified to implement C code by Dave Benson.
 
+#include <sstream>
+// Yes, this is evil. You're free to fix it. Please share the patch.
+#define private public
+#include <google/protobuf/descriptor.h>
+#undef private
+
 #include <protoc-c/c_generator.h>
 
 #include <memory>
diff -up protobuf-c-1.4.1/protoc-c/c_helpers.h.omv~ protobuf-c-1.4.1/protoc-c/c_helpers.h
